Algorithms/Ruby/min-depth-binary-tree.rb

```diff
@@ -0,0 +1,20 @@
+# Definition for a binary tree node.
+# class TreeNode
+#     attr_accessor :val, :left, :right
+#     def initialize(val)
+#         @val = val
+#         @left, @right = nil, nil
+#     end
+# end
+
+# @param {TreeNode} root
+# @return {Integer}
+
+def min_depth_binary_tree(root)
+  return 0 unless root
+  return 1 unless root.left && root.right
+  return 1 + min_depth_binary_tree(root.left) unless root.right
+  return 1 + min_depth_binary_tree(root.right) unless root.left
```

Comment:
`unless root.left && root.right` == `if !(root.left && root.right)` == `if !root.left  || !root.right` == `if one of these are false`, so the following lines will never fire.